cmd/admin-handlers-users.go

```diff
@@ -540,33 +545,29 @@ func (a adminAPIHandlers) AddServiceAccount(w http.ResponseWriter, r *http.Reque
 	}
 
 	var ldapUsername string
-	if globalLDAPConfig.Enabled && targetUser != "" {
-		// If LDAP enabled, service accounts need
-		// to be created only for LDAP users.
+	switch {
+	case owner && targetUser == cred.AccessKey:
+		// Allow service account creation for root user.
+	case globalLDAPConfig.Enabled:
+		// If LDAP enabled, service accounts need to be created only for
+		// LDAP users.
 		var err error
 		ldapUsername = targetUser
 		targetUser, targetGroups, err = globalLDAPConfig.LookupUserDN(targetUser)
 		if err != nil {
 			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
 			return
 		}
-		// targerUser is set to bindDN at this point in time.
-		// targetGroups is set to the groups at this point in time.
-	} else {
+	case targetUser == cred.AccessKey:
```

Comment:
We are missing another use case here and I think it is already working. Which is allowing root or any user that has enough permissions to create a service account for another user.

Reply:
It seems to be working fine - with a server built on this PR:

```
$ mc admin user add myminio bar barfoo123
Added user `bar` successfully.
$ mc admin policy set myminio consoleAdmin user=bar
Policy `consoleAdmin` is set on user `bar`
$ MC_HOST_bar=http://bar:barfoo123@localhost:9000 mc mb bar/test2
Bucket created successfully `bar/test2`.
$ mc admin user svcacct add myminio bar --access-key barsvc1 --secret-key barsvc1123
Added service account `barsvc1` successfully.
$ MC_HOST_barsvc1=http://barsvc1:barsvc1123@localhost:9000 mc mb barsvc1/test3
Bucket created successfully `barsvc1/test3`.
$ MC_HOST_barsvc1=http://barsvc1:barsvc1123@localhost:9000 mc ls barsvc1
[2021-09-17 12:13:44 PDT]     0B test/
[2021-09-17 12:17:38 PDT]     0B test2/
[2021-09-17 12:19:42 PDT]     0B test3/
```

We are checking if the user has permission to create the service account before this switch block. Validations for the target-user of the service account (e.g. should not already be a service account) are being done in `IAMSys.NewServiceAccount()` method.

Previous to this change, when `targetUser != cred.AccessKey`, we are validating that `cred.ParentUser` is not empty - this is not really necessary as the parent user for the service account to be created would be `targetUser` anyway.